### ISSUE TITLE
Processing configuration for http auth (environment variables)

### ISSUE THREAD

**Issue by fralik** [python-poetry/poetry#1871]
<!-- Checked checkbox should look like this: [x] -->
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10, 64-bit.
- **Poetry version**: 1.0.2

## Issue
Documentation [says](https://python-poetry.org/docs/configuration/#using-environment-variables) that any config value can be passed through an environment variable `POETRY_<config>` as well as secrets for private repos. However, I suspect that the code doesn't process secrets correctly.

**Setup:**
1. There is no `auth.toml` file.
2. Url for the private repo is added in `config.toml`.
3. `pyproject.toml` contains url and name for the same private repo.
4. `pyproject.toml` references one package from the private repo.
5. Variables `POETRY_HTTP_BASIC_REPO_PASSWORD` and `POETRY_HTTP_BASIC_REPO_USERNAME` are set.
6. Variable `POETRY_HTTP_BASIC_REPO` is not set.
7. Some debug output (`print`) is added to poetry source code.

**Outcome:**
```
>poetry install --dry-run --no-ansi -vvv                                                                        
Getting auth for repo
Need to get setting http-basic.repo
Checking variable POETRY_HTTP_BASIC_REPO, value <object object at 0x00000239FFCD5900>, not set <object object at 0x00000239FFCD5900>
Value is not set, so continue
Checking key: http-basic
Will return default: None
Failed to get auth                                                                                                                                            
Using virtualenv: c:\home\anaconda3\envs\env-poetry                                                                                                           
Updating dependencies                                                                                                                                         
Resolving dependencies...                                                                                                                                     
   1: fact: demo is 0.1.0                                                                                                                           
   1: derived: demo                                                                                                                                 
   1: fact: demo depends on treeview (~0.1.0)                                                                                                       
   1: selecting demo (0.1.0)                                                                                                                        
   1: derived: treeview (~0.1.0)                                                                                                                              
   1: fact: no versions of treeview match >=0.1.0,<0.2.0                                                                                                      
   1: conflict: no versions of treeview match >=0.1.0,<0.2.0                                                                                                  
   1: ! treeview (~0.1.0) is satisfied by treeview (~0.1.0)                                                                                                   
   1: ! which is caused by "demo depends on treeview (~0.1.0)"                                                                                      
   1: ! thus: version solving failed                                                                                                                          
   1: Version solving took 0.348 seconds.                                                                                                                     
   1: Tried 1 solutions.                                                                                                                                      
                                                                                                                                                              
[SolverProblemError]                                                                                                                                          
Because demo depends on treeview (~0.1.0) which doesn't match any versions, version solving failed.
```

**Thoughts:**
It looks like [get_http_auth](https://github.com/python-poetry/poetry/blob/636ce8b0eba7dfa390b3fd961d1b9fb533d5d033/poetry/utils/password_manager.py#L155) expects a dictionary. We can only get a dictionary if we read from the config file, not the environment variable. Or am I wrong here?

I made more experiments around this (like defining `POETRY_HTTP_BASIC_REPO` with various content), but couldn't get around the fact that `auth` must be a dictionary.

Variables `POETRY_HTTP_BASIC_REPO_USERNAME` and `POETRY_HTTP_BASIC_REPO_PASSWORD` seem never to be read. Only if I reference them with smth like this:
```
export POETRY_HTTP_BASIC_REPO={http.basic.repo.username} {http.basic.repo.password}
```
but this gets me nowhere as this is again not a dictionary.

## Conclusion
It seems to me that either the code or documentation shall be changed. My suspicion is that the code follows your design and you won't want to change it. So perhaps the documentation shall be a bit more clear about such usage.

This issue is somewhat related to #208 .

**Comment by sztomi** [comment]
I just ran into this. The documentation clearly states this fact in multiple places and this is referenced also in PRs/issues connected to private repositories. A usable workaround is writing the config each time via `poetry config http-basic.<reponame> <user> <password>` where `reponame` is a repository previously configured.

**Comment by jacek-jablonski** [comment]
I can confirm this bug.

Temporary workaround would be setting
`export POETRY_HTTP_BASIC_REPO=`
and then `POETRY_HTTP_BASIC_REPO_USERNAME` and `POETRY_HTTP_BASIC_REPO_PASSWORD` are read.

**Comment by janjagusch** [comment]
I cannot reproduce this issue anymore with the most recent version of poetry (v1.0.9).
Can anybody confirm that this has been fixed?

**Comment by jooh** [comment]
> I can confirm this bug.
> 
> Temporary workaround would be setting `export POETRY_HTTP_BASIC_REPO=` and then `POETRY_HTTP_BASIC_REPO_USERNAME` and `POETRY_HTTP_BASIC_REPO_PASSWORD` are read.

Can confirm this bug persists on poetry 1.1.12, and this workaround is effective

**Comment by neersighted** [comment]
This is solved on the 1.2 branch.
